### PR TITLE
Dependency Changes

### PR DESCRIPTION
**Pull request by Sharpz7**
This includes:

- [ ] https://github.com/armadaproject/armada/pull/2949
- [ ] https://github.com/armadaproject/armada/pull/2945
- [ ] https://github.com/armadaproject/armada/pull/2944
- [ ] https://github.com/armadaproject/armada/pull/2943
- [ ] https://github.com/armadaproject/armada/pull/2942
- [ ] https://github.com/armadaproject/armada/pull/2941
- [ ] https://github.com/armadaproject/armada/pull/2930
- [ ] https://github.com/armadaproject/armada/pull/2929
- [ ] https://github.com/armadaproject/armada/pull/2928
- [ ] https://github.com/armadaproject/armada/pull/2927
- [ ] https://github.com/armadaproject/armada/pull/2926

### CODE REVIEW

**Comment by Sharpz7** [comment]
To the best of my knowledge, the bump in sphinx version is not possible as long as Python 3.8 is required.

File: internal/common/pgkeyvalue/pgkeyvalue_test.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"golang.org/x/exp/maps"
-	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/utils/clock"
```
Comment (Sharpz7) [inline internal/common/pgkeyvalue/pgkeyvalue_test.go:12]: This doesn't seem to be the right change, but something like this needs to happen. I got errors when using the apimachinery code for it.

**Comment by Sharpz7** [comment]
No idea how to fix that proto error. @dejanzele any suggestions?

**Comment by kannon92** [comment]
Tbh you probably should ignore any K8s updates.  I’d drop the commit for kubelet.

**Comment by kannon92** [comment]
And just close the kube updates. This project. Is pretty far from the supported versions of kube but it would be a lot of work to update the apis to 1.28 (from 1.22)

**Comment by Sharpz7** [comment]
Hey @kannon92 

The problem is always that closed ones will re-appear. I am going to close this PR and start again, without the sphinx and kube PR's

**Pull request closed without merging**